src/test/java/ui/RegisterTest.java

```diff
@@ -26,20 +27,29 @@ public RegisterTest() {
         this.password = getPropertyFromRegProp("password");
     }
 
-
+    public static String getPropertyFromRegProp(String prop){
+        return getPropertyValue(getPropertyFileContent(REG_FILE), prop);
+    }
 
     @BeforeMethod
     void beforeMethod() {
         this.registerPage = new RegisterPage(getDriver());
     }
 
+    @BeforeMethod
+    void beforeMethodForWordpress() {
+        this.wordpressPage = new WordpressPage(getDriver());
+    }
+
     @Test
     public void testSuccessfulRegistration() {
         this.registerPage.open();
         this.registerPage.verify();
         this.registerPage.register(username, email, password);
-        String validationMessage = this.registerPage.getRegisterMessage();
-        Assert.assertEquals(validationMessage, this.registerPage.getExpectedRegisterMsg(), "The error message is " +
+
+        this.wordpressPage.open();
+        String validationMessage = this.wordpressPage.getRegisterMessage();
+        Assert.assertEquals(validationMessage, this.wordpressPage.getExpectedRegisterMsg(), "The error message is " +
```

Comment:
The registration flow is complete only after you fill in again the credentials in the wordpress page and you are redirected back to the shop.demoqa page. So the assert of this test (successful registration flow) is to validate that the message from my account page (page displayed after performing the above steps) says "Hello, username..".
Asserting the message from the wordpress page could be a different test.

Reply:
Done